src/NServiceBus.Metrics.ServiceControl/MetricsReportBaseHeaderFactory.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace NServiceBus.Metrics.ServiceControl
+{
+    using System.Collections.Generic;
+    using Hosting;
+    using Support;
+
+    class MetricsReportBaseHeaderFactory
+    {
+        public MetricsReportBaseHeaderFactory(string endpointName, ReportingOptions options)
+        {
+            this.endpointName = endpointName;
+            this.options = options;
+        }
+
+        internal Dictionary<string, string> BuildBaseHeaders(HostInformation hostInformation)
+        {
+            var headers = new Dictionary<string, string>
+            {
+                {Headers.OriginatingEndpoint, endpointName},
+                {Headers.OriginatingMachine, RuntimeEnvironment.MachineName},
+                {Headers.OriginatingHostId, hostInformation.HostId.ToString("N")},
+                {Headers.HostDisplayName, hostInformation.DisplayName},
```

Comment:
Missing a session ID, although less relevant I think it could still be valuable to be able to differentiate messages send from different instances.

Reply:
@mikeminutillo Runtime, how else can we reset the count or differentiate between multiple instances? I know that with native queue length last writer wins but I think it is *always* a good thing to add.

Reply:
The way this currently works is that the Max value for the interval wins. But that is handled at the server end. Why do we need to reset the count? If you restart an instance, it still has the same queue length as the previous instance.